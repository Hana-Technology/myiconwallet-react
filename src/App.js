import React from 'react';
import logo from './logo.svg';

function App() {
  return (
    <div className="text-center">
      <header className="bg-gray-800 text-white text-sm sm:text-base md:text-lg lg:text-xl min-h-screen flex flex-col items-center justify-center">
        <img src={logo} style={{ height: '40vmin' }} alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="text-teal-400"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default App;
