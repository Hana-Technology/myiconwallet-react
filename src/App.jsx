import React from 'react';
import Logo from 'components/Logo';

function App() {
  return (
    <div className="flex flex-col min-h-screen">
      <header className="bg-gray-800 text-gray-100 p-4 pt-3 shadow-lg">
        <div className="container mx-auto">
          <h1 className="text-2xl lg:text-3xl whitespace-no-wrap" aria-label="my icon wallet">
            my
            <Logo className="text-cyan mx-1" />
            wallet
          </h1>
        </div>
      </header>
      <main className="flex-1 text-gray-800 py-6">
        <div className="container mx-auto">
          <p className="text-lg">
            Your <Logo iconClassName="text-teal-600" /> wallet goes here!
          </p>
        </div>
      </main>
      <footer className="bg-gray-200 text-gray-600 p-4">
        <div className="container mx-auto text-right text-sm">
          Â© {new Date().getFullYear()}{' '}
          <a className="hover:text-gray-700" href="https://www.reliantnode.com/">
            ReliantNode
          </a>
        </div>
      </footer>
    </div>
  );
}

export default App;
