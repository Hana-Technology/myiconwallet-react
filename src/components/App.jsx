import React from 'react';
import { Router } from '@reach/router';
import IconService from 'components/IconService';
import Wallet from 'components/Wallet';
import Home from 'pages/Home';
import CreateWallet from 'pages/CreateWallet';
import UnlockWallet from 'pages/UnlockWallet';
import Stake from 'pages/Stake';
import Vote from 'pages/Vote';
import NotFound from 'pages/NotFound';

function App() {
  return (
    <IconService>
      <Wallet>
        <Router>
          <Home path="/" />
          <CreateWallet path="/create" />
          <UnlockWallet path="/unlock" />
          <Stake path="/stake" />
          <Vote path="/vote" />
          <NotFound default />
        </Router>
      </Wallet>
    </IconService>
  );
}

export default App;
