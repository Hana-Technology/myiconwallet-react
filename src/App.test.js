import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

const CURRENT_YEAR = new Date().getFullYear();

it('renders the header and footer', () => {
  const { getByText } = render(<App />);
  expect(getByText(/my.*wallet/)).toMatchSnapshot();
  expect(getByText(`© ${CURRENT_YEAR}`)).toHaveTextContent(`© ${CURRENT_YEAR} ReliantNode`);
});
