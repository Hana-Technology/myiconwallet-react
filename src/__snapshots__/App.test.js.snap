// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the header and footer 1`] = `
<h1
  aria-label="my icon wallet"
  class="text-2xl lg:text-3xl whitespace-no-wrap"
>
  my
  <svg
    aria-label="ICON Foundation logo"
    class="fill-current inline align-middle text-cyan mx-1"
    data-testid="logo"
    role="logo"
    style="height: 0.675em;"
    viewBox="0 0 350 120"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      data-testid="logo-icon"
    >
      <path
        d="M233.5 31.4l-13 13c2.8 4.6 4.4 9.9 4.4 15.7 0 16.6-13.4 30-30 30-5.7 0-11.1-1.6-15.7-4.4l-13 13c8 5.9 17.9 9.5 28.6 9.5 26.5 0 48-21.5 48-48 .2-10.9-3.3-20.8-9.3-28.8zM165 60c0-16.6 13.4-30 30-30 5.7 0 11.1 1.6 15.7 4.4l13-13c-8-5.9-17.9-9.5-28.6-9.5-26.5 0-48 21.5-48 48 0 10.7 3.5 20.6 9.5 28.6l13-13c-3-4.4-4.6-9.8-4.6-15.5z"
      />
      <circle
        cx="243"
        cy="12"
        r="12"
      />
      <circle
        cx="147"
        cy="108"
        r="12"
      />
    </g>
    <g
      data-testid="logo-text"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
      <path
        d="M20.5 106h-17c-.3 0-.5-.2-.5-.5v-73c0-.3.2-.5.5-.5h17c.3 0 .5.2.5.5v73c0 .3-.2.5-.5.5zM90.6 32h28.9c.3 0 .5-.2.5-.5v-17c0-.3-.2-.5-.5-.5H90.8C65.6 14 44.5 34 44 59.2 43.6 85 64.3 106 90 106h29.5c.3 0 .5-.2.5-.5v-17c0-.3-.2-.5-.5-.5H90c-15.7 0-28.4-13-28-28.8C62.4 44 75.3 32 90.6 32zM306.3 14c-23.6.4-42.3 20.1-42.3 43.7v47.8c0 .3.2.5.5.5h17c.3 0 .5-.2.5-.5V57.6c0-14 11.5-25.8 25.5-25.6 13.6.3 24.5 11.3 24.5 25v48.5c0 .3.2.5.5.5h17c.3 0 .5-.2.5-.5V57c0-24-19.7-43.4-43.7-43z"
      />
    </g>
  </svg>
  wallet
</h1>
`;
